lnrpc/rpc.proto

```diff
@@ -2909,12 +2909,39 @@ message ListPaymentsRequest {
     that pending payments, as well as failed payments will show up if this
     field is set to True.
     */
-    bool include_incomplete = 1;
+    bool include_incomplete = 1 [json_name = "include_incomplete"];
+
+    /**
+    The index of a payment that will be used as either the start or end of a
+    query to determine which payments should be returned in the response.
+    */
+    uint64 index_offset = 2 [json_name = "index_offset"];
+
+    /// The max number of payments to return in the response to this query.
+    uint64 num_max_payments = 3 [json_name = "num_max_payments"];
+
+    /**
+    If set, the payments returned will result from seeking backwards from the
+    specified index offset. This can be used to paginate backwards.
+    */
+    bool reversed = 4 [json_name = "reversed"];
 }
 
 message ListPaymentsResponse {
     /// The list of payments
-    repeated Payment payments = 1;
+    repeated Payment payments = 1 [json_name = "payments"];
+
+    /**
+    The index of the first item in the set of returned payments. This can be
+    used to seek backwards, pagination style.
+    */
+    uint64 first_index_offset = 2 [json_name = "first_index_offset"];
```

Comment:
using both the words `index` and `offset`, isn't that redundant? Maybe `first_index` and `last_index` is sufficient

Reply:
It's again due compatibility with listinvoices, have also thought about that. It's an offset for an index, which is ok, I think.